src/care/host_device_ptr.h

```diff
@@ -456,14 +456,6 @@ namespace care {
       CARE_HOST void move(ExecutionSpace space) {
          MA::move(chai::ExecutionSpace((int) space));
       }
-      
-      CARE_HOST_DEVICE inline bool operator ==(host_device_ptr<T, Accessor> const & right) const {
```

Comment:
Alternatively, I could move this outside the class and add nullptr overloads.

Reply:
The CHAI version only compares the active pointers, but I think we enforce consistency enough that that should be sufficient. And most often we just compare against nullptr.

Reply:
To be honest, I'm not convinced either ours or CHAI's implementation is really correct. Perhaps we ought to be comparing the pointer records.

Reply:
I'll approve this MR in any case, but yes something to discuss with in the future